File: CoreJava/doug's-dog-doors-app/src/components/Bark.java
package components;

public class Bark {
	private String sound;
	
	public Bark(String sound){
		this.sound=sound;
	}
	
	public String getSound(){
		return sound;
	}
	
	public boolean equals(Object bark){
		//if(){
			Bark otherBark= (Bark) bark;
			if(this.sound.equalsIgnoreCase(otherBark.sound)){
				return true;
			}
		//}
		return false;
	}
}

File: C#/OOP/DogDoorApp/DogDoorApp/Bark.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace DogDoorApp
{
    class Bark
    {
        private string _sound;

        public Bark(string sound)
        {
            _sound = sound;
        }

        public string Sound
        {
            get
            {
                return _sound;
            }
        }

        public bool EqualsBark(object bark)
        {
            if (typeof(Bark) == bark.GetType())
            {
                Bark otherBark = (Bark)bark;
                if (this._sound.Equals(otherBark))
                    return true;
            }
            return false;
        }
    }
}
